data-serving/data-service/src/controllers/preprocessor.ts

```diff
@@ -119,6 +123,30 @@ export const setBatchUpsertRevisionMetadata = async (
     next();
 };
 
+export const findCasesToUpdate = async (
+    request: Request,
+    response: Response,
+    next: NextFunction,
+) => {
+    // Find all cases matching the query
+    const matchedCases = await (casesMatchingSearchQuery({
+        searchQuery: request.body.query,
+        count: false,
+    }) as DocumentQuery<CaseDocument[], CaseDocument, {}>).exec();
+
+    // Set those case ids to be updated with the request case.
+    const casesToUpdate = matchedCases.map((c: any) => {
+        return { _id: c._id, ...request.body.case };
+    });
+    request.body.cases = casesToUpdate;
+
+    // Delete no longer used fields
+    delete request.body.query;
+    delete request.body.case;
```

Comment:
Nice job reusing existing middleware. That said, (I digress, but) I wish we didn't need to mess with metadata/revisions for these -- would be much simpler and more efficient.